File: website/src/main/java/org/bag/Application.java
package org.bag;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.ComponentScan;
import org.springframework.web.servlet.LocaleResolver;
import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
import org.springframework.web.servlet.i18n.LocaleChangeInterceptor;
import org.springframework.web.servlet.i18n.SessionLocaleResolver;

import java.util.Locale;

@SpringBootApplication
@ComponentScan("org.bag.controllers")
@ComponentScan("org.bag.service")
public class Application implements WebMvcConfigurer {

    public static void main(String[] args) {
        Locale.Builder builder = new Locale.Builder().setLanguage("nl").setRegion("US");
        Locale locale1 = builder.build();
        String iso3Country = locale1.getISO3Country();
        String iso3Language = locale1.getISO3Language();

        SpringApplication.run(Application.class, args);
    }

    @Bean
    public LocaleResolver localeResolver() {
        SessionLocaleResolver slr = new SessionLocaleResolver();
        slr.setDefaultLocale(new Locale("nl","NL"));
        return slr;
    }

    @Bean
    public LocaleChangeInterceptor localeChangeInterceptor() {
        LocaleChangeInterceptor lci = new LocaleChangeInterceptor();
        lci.setParamName("lang");
        return lci;
    }

    @Override
    public void addInterceptors(InterceptorRegistry registry) {
        registry.addInterceptor(localeChangeInterceptor());
    }
}

File: website/src/main/kotlin/org/bag/Application.kt
package org.bag

import com.zaxxer.hikari.HikariConfig
import com.zaxxer.hikari.HikariDataSource
import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.ComponentScan
import org.springframework.context.annotation.ComponentScans
import org.springframework.web.servlet.LocaleResolver
import org.springframework.web.servlet.config.annotation.InterceptorRegistry
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer
import org.springframework.web.servlet.i18n.LocaleChangeInterceptor
import org.springframework.web.servlet.i18n.SessionLocaleResolver

import java.util.Locale
import javax.sql.DataSource

@SpringBootApplication
@ComponentScans(
        ComponentScan("org.bag.controllers"),
        ComponentScan("org.bag.service"))
open class Application : WebMvcConfigurer {

    @Bean
    open fun localeResolver(): LocaleResolver {
        val slr = SessionLocaleResolver()
        slr.setDefaultLocale(Locale("nl", "NL"))
        return slr
    }

    @Bean
    open fun localeChangeInterceptor(): LocaleChangeInterceptor {
        val lci = LocaleChangeInterceptor()
        lci.paramName = "lang"
        return lci
    }

    @Bean
    open fun dataSource(): DataSource {
        val config = createHikariConfig()
        return HikariDataSource(config)
    }

    private fun createHikariConfig(): HikariConfig {
        val config = HikariConfig()

        // some comment to deploy

        // Is er een heroku cs?
        val connectionString = System.getenv()["JDBC_DATABASE_URL"]
        if (connectionString == null) {
            // Nee, connectie maken met kantoor
            config.jdbcUrl = "jdbc:postgresql://127.0.0.1:5432/bag-java"
            config.username = "postgres"
            config.password = "toor"
        } else {
            // Ja, die gebruiken
            config.jdbcUrl = connectionString
        }

        return config
    }

    override fun addInterceptors(registry: InterceptorRegistry) {
        registry.addInterceptor(localeChangeInterceptor())
    }
}

fun main(args: Array<String>) {
    Locale.setDefault(Locale("nl", "NL"))
    SpringApplication.run(Application::class.java, *args)
}


